core/domain/feedback_services_test.py

```diff
@@ -268,59 +265,114 @@ def setUp(self):
+        feedback_services.add_message_to_email_buffer(
+            self.user_id_a, self.exploration.id, self.thread_id,
+            self.message_id1)
         self.assertEqual(self.count_jobs_in_taskqueue(), 1)
 
         expected_feedback_message_dict = {
-            'exploration_id': self.exp_id,
+            'exploration_id': self.exploration.id,
             'thread_id': self.thread_id,
             'message_id': self.message_id1
         }
-        model = feedback_models.UnsentFeedbackEmailModel.get(self.user_id_a)
+        model = feedback_models.UnsentFeedbackEmailModel.get(self.editor_id)
         self.assertDictEqual(
             model.feedback_message_references[0],
             expected_feedback_message_dict)
         self.assertEqual(model.retries, 0)
-        self.process_and_flush_pending_tasks()
 
     def test_add_new_feedback_message(self):
-        exp = exp_domain.Exploration.create_default_exploration(self.exp_id)
-        exp_services.save_new_exploration(self.user_id_a, exp)
-
-        feedback_services.send_feedback_message_email(
-            self.exp_id, self.thread_id, self.message_id1)
-        feedback_services.send_feedback_message_email(
-            self.exp_id, self.thread_id, self.message_id2)
+        feedback_services.add_message_to_email_buffer(
+            self.user_id_a, self.exploration.id, self.thread_id,
+            self.message_id1)
+        feedback_services.add_message_to_email_buffer(
+            self.user_id_a, self.exploration.id, self.thread_id,
+            self.message_id2)
 
         self.assertEqual(self.count_jobs_in_taskqueue(), 1)
 
         expected_feedback_message_dict1 = {
-            'exploration_id': self.exp_id,
+            'exploration_id': self.exploration.id,
             'thread_id': self.thread_id,
             'message_id': self.message_id1
         }
         expected_feedback_message_dict2 = {
-            'exploration_id': self.exp_id,
+            'exploration_id': self.exploration.id,
             'thread_id': self.thread_id,
             'message_id': self.message_id2
         }
 
-        model = feedback_models.UnsentFeedbackEmailModel.get(self.user_id_a)
+        model = feedback_models.UnsentFeedbackEmailModel.get(self.editor_id)
         self.assertDictEqual(
             model.feedback_message_references[0],
             expected_feedback_message_dict1)
         self.assertDictEqual(
             model.feedback_message_references[1],
             expected_feedback_message_dict2)
         self.assertEqual(model.retries, 0)
-        self.process_and_flush_pending_tasks()
+
+    def test_that_emails_are_not_sent_for_anonymous_user(self):
+        with self.can_send_emails_ctx, self.can_send_feedback_email_ctx:
+            feedback_services.create_thread(
+                self.exploration.id, 'a_state_name', None, 'a subject',
+                'some text')
+
+            self.assertEqual(self.count_jobs_in_taskqueue(), 1)
```

Comment:
It is 1 because when we use create_thread() function, it executes one MapReduce task and this 1 is for that MapReduce task. We can not assert tasks[0].url because we do not have any task in taskqueue of emails. This is test for that email is not being sent and if test fails then we will have 2 pending tasks in queue. 
About comment, we will have to add it to all test units, because each of them uses this function. What comment string you suggest ? I am sure it will be better than what I would write.